src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerEnclaveProvider.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.util.ArrayList;
+
+
+public interface ISQLServerEnclaveProvider {
+    byte[] getEnclavePackage(String userSQL, ArrayList<byte[]> enclaveCEKs) throws SQLServerException;
+
+    void getAttestationParamters(boolean b,

```

Comment:
Will all implementations of this interface throw the same exceptions? If not, this is implementation specific and needs to be handled in the class.